File: 1624 largest-substring-between-two-equal-characters/solution.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
    int[] arr=new int[26];
    Arrays.fill(arr,-1);
    int max=-1;
    for(int i=0;i<s.length();i++){
        int idx=s.charAt(i)-'a';
        if(arr[idx]==-1) arr[idx]=i;
        else max=Math.max(max,i-arr[idx]-1);
    }
    return max;
}
}

File: 1624 largest-substring-between-two-equal-characters/solution.go
func maxLengthBetweenEqualCharacters(s string) int {
    f:=make([]int,26)
    for i:= range f {
        f[i]=-1
    }
    ma:=-1
    for i,v := range s {
        if f[v-'a']==-1 {
            f[v-'a'] = i
        } else {
            ma = max(ma,i-f[v-'a']-1)
        }
    }
    return ma
}
